ts/features/bonus/cgn/screens/merchants/CgnMerchantsCategoriesSelectionScreen.tsx

```diff
@@ -0,0 +1,137 @@
+import customVariables from "../../../../../theme/variables";
+import { ProductCategoryEnum } from "../../../../../../definitions/cgn/merchants/ProductCategory";
+import TouchableDefaultOpacity from "../../../../../components/TouchableDefaultOpacity";
+import { H2 } from "../../../../../components/core/typography/H2";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { useIODispatch } from "../../../../../store/hooks";
+import { useNavigationContext } from "../../../../../utils/hooks/useOnFocus";
+import CGN_ROUTES from "../../navigation/routes";
+import { cgnSelectedCategory } from "../../store/actions/categories";
+import { H1 } from "../../../../../components/core/typography/H1";
+import { EdgeBorderComponent } from "../../../../../components/screens/EdgeBorderComponent";
+
+const styles = StyleSheet.create({
+  body: {
+    borderRadius: 8,
+    flex: 1,
+    marginBottom: 10,
+    marginRight: widthPercentageToDP("2.93%"),
+    width: widthPercentageToDP("42.13%")
+  },
+  container: {
+    flexDirection: "column",
+    paddingHorizontal: customVariables.contentPadding / 2,
+    paddingVertical: 14
+  }
+});
+
+const CgnMerchantsCategoriesSelectionScreen = () => {
+  const dispatch = useIODispatch();
+  const navigation = useNavigationContext();
+
+  const renderCategoryElement = (
+    info: ListRenderItemInfo<ProductCategoryEnum | "All" | "Hidden">
+  ) => {
+    if (info.item === "All") {
+      return (
+        <LinearGradient
+          colors={["#475A6D", "#E6E9F2"]}
+          useAngle={true}
+          angle={57.23}
+          style={styles.body}
+        >
+          <TouchableDefaultOpacity
+            style={[IOStyles.flex, styles.container]}
+            onPress={() =>
+              navigation.navigate(CGN_ROUTES.DETAILS.MERCHANTS.LIST)
+            }
+          >
+            <View style={[IOStyles.flex, IOStyles.row]}>
+              <H2 color={"white"}>
+                {I18n.t("bonus.cgn.merchantDetail.categories.all")}
+              </H2>
+            </View>
+          </TouchableDefaultOpacity>
+        </LinearGradient>
+      );
+    }
+    if (info.item === "Hidden") {
+      return (
+        <View style={[styles.body, { height: 122 }]}>
+          <View style={[IOStyles.flex, styles.container]} />
+        </View>
+      );
+    }
+    const specs = getCategorySpecs(info.item);
+
+    return specs.fold(null, s => (
+      <LinearGradient
+        colors={s.colors}
+        useAngle={true}
+        angle={57.23}
+        style={styles.body}
+      >
+        <TouchableDefaultOpacity
+          style={[IOStyles.flex, styles.container]}
+          onPress={() => {
+            dispatch(cgnSelectedCategory(s.type));
+            navigation.navigate(CGN_ROUTES.DETAILS.MERCHANTS.LIST_BY_CATEGORY);
+          }}
+        >
+          <View style={[IOStyles.flex, IOStyles.row]}>
+            <H2 color={"white"}>{I18n.t(s.nameKey)}</H2>
```

Comment:
I'm pretty sure this should be impossible, since the nameKey is defined in the category specs collection as a `TranslationKey` if a category exists its nameKey exists or at least we should have a compile/lint error... what do you think?

Reply:
Ops you're right, I didn't see that `nameKey` is a `TranslationKeys`, so we don't need any fallback here!